### DISCUSSION TITLE
Greenfield API : PoS App get/add/update/remove individual items

### DISCUSSION THREAD

**Discussion by Atitlanio** [btcpayserver/btcpayserver#4652] [Ideas / Feature Requests]
So we now have the ability to interact with the Greenfield API and retrieve **_all_** of the items of a PoS app following the commit:  Greenfield get app details (https://github.com/btcpayserver/btcpayserver/pull/4102) @bolatovumar 

It seems like a natural progression to  have API control over **_individual_** items of a PoS app. 

For example, at the moment in order to update items it requires a PUT [link](https://docs.btcpayserver.org/API/Greenfield/v1/#operation/Apps_PutPointOfSaleApp) using a clunky `template` field that requires including the information for the totality of items in the app, e.g.,:
```
{"appName": "Store Name",
 "template": "{ 'First Item Name': {price: 30, description: 'hello', image: https://picsum.photos/300/300, price_type: fixed, disabled: true, inventory: 3}, "Second Item Name": {price: 30, description: 'hello', image: https://picsum.photos/300/300, price_type: fixed, disabled: true, inventory: 3}}",},
...
```

(Unless I am doing something wrong, the above does not allow for specification of the `id` of the item, it simply takes on, e.g., 'First Item Name' for the `title` as well as the `id`)

Presumably it would look something like :
POST : require `appId` 
GET/PUT/DELETE : require the `appId` and  `itemId` 

PUT/POST Query Parameters:
            title: string
            description: string,
            id: string,
            image: string,
            price: type: ['Fixed', 'Minimum', 'Custom'] ; value: int
            buyButtonText: string,
            inventory: int,
            paymentMethods: ?,
            disabled: bool
   
The application I would like to use this for is an online market place with a custom front end. That way individual items can be modified through the API, e.g., if updating the inventory while an item is waiting confirmation on the blockchain.

**Comment by bolatovumar** [comment]
@NicolasDorier @Kukks @pavlenex thoughts on this one?
